src/rai_asr/rai_asr/asr_node.py

```diff
@@ -153,80 +254,120 @@ def hmi_status_callback(self, msg: String):
+        elif self.is_recording:
+            self.audio_buffer.append(indata)
+            if not isinstance(self.silence_start_time, float):
+                raise ValueError(
+                    "Silence start time is not set, this should not happen"
+                )
+            if time.time() - self.silence_start_time > self.silence_grace_period:
+                self.stop_recording_and_transcribe()
+
+    def initialize_sounddevice_stream(self):
+        sd.default.latency = ("low", "low")
+        self.device_sample_rate = sd.query_devices(
+            device=self.recording_device_number, kind="input"
+        )[
+            "default_samplerate"
+        ]  # type: ignore
+        self.window_size_samples = int(
+            DEFAULT_BLOCKSIZE * self.device_sample_rate / DEFAULT_SAMPLING_RATE
+        )
+        self.stream = sd.InputStream(
+            samplerate=self.device_sample_rate,
             channels=1,
             device=self.recording_device_number,
             dtype="int16",
-            blocksize=window_size_samples,
+            blocksize=self.window_size_samples,
+            callback=self.sd_callback,
         )
-        stream.start()
-        self.get_logger().info(
-            "Voice Activity Detection enabled. Waiting for speech..."
-        )
-        while True:
-            audio_data, _ = stream.read(window_size_samples)
-            audio_data = audio_data.flatten()
+        self.stream.start()
 
-            asr_lock = (
-                time.time()
-                < self.last_transcription_time + self.transcription_recording_timeout
-            )
-            if asr_lock or self.hmi_lock or self.tts_lock:
-                continue
-
-            speech_dict = self.vad_iterator(audio_data, return_seconds=True)
-            if speech_dict:
-                if not self.is_recording:
-                    self.start_recording()
-                self.audio_buffer.append(audio_data)
-                self.silence_start_time = None
-                if "end" in speech_dict.keys():
-                    self.silence_start_time = datetime.now()
-            elif self.is_recording:
-                self.audio_buffer.append(audio_data)
-                if self.silence_start_time is not None:
-                    if datetime.now() - self.silence_start_time > self.grace_period:
-                        self.stop_recording_and_transcribe()
+    def reset_buffer(self):
+        self.audio_buffer.clear()
 
     def start_recording(self):
-        self.get_logger().info("Recording...")
+        self.get_logger().info("Recording...")  # type: ignore
         self.publish_status("recording")
         self.is_recording = True
-        self.audio_buffer = []
-        self.silence_start_time = None
 
     def stop_recording_and_transcribe(self):
-        self.get_logger().info("Stopped recording. Transcribing...")
+        self.get_logger().info("Stopped recording. Transcribing...")  # type: ignore
```

Comment:
Can't see type conflict here, did it break `flake8` for you?
```suggestion
        self.get_logger().info("Stopped recording. Transcribing...")
```